src/main/java/ghostcat/loginLambda/Login.java

```diff
@@ -0,0 +1,31 @@
+package ghostcat.loginLambda;
+
+import java.security.SecureRandom;
+import java.time.Instant;
+
+public class Login {
+  public LoginResult handleRequest(LoginRequest request) {
+    LoginResult result = new LoginResult();
+    LoginDAO dao = new LoginDAO();
+
+    if (dao.valid(request.getUserID(), request.getPasswordHash())) {
+      String AB = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+      SecureRandom rnd = new SecureRandom();
+      StringBuilder stringBuilder = new StringBuilder();
+      for (int i = 0; i < 10; ++i) {
+        stringBuilder.append( AB.charAt( rnd.nextInt(AB.length()) ) );
+      }
+      StringBuilder time = new StringBuilder();
+      time.append(Instant.now().toEpochMilli());
+      stringBuilder.append(time.toString());
```

Comment:
This really doesn't matter and is just an FYI, but most hashing algorithms work on chunks of the input sequentially, so it's generally best practice to prepend the salt instead of appending it. (for this example: `time.append(stringBuilder.toString())`)

Seriously though I doubt it matters on this project so no worries about changing it, just wanted to leave the comment as a FYI more than anything.

Reply:
### Ignore my previous comment ^

Sorry, I hadn't finished reading through the code. The time and string builder variables here are just being used for the auth token, not as a salt. It looks like we're not using salts, so all of the salt talk in the comment above can be ignored

Reply:
No worries, I was thinking of encrypting the password before sending it to the API, so this lambda doesn't encrypt or anything.